component/controller/SerialPortBridgeGeneric.cpp

```diff
@@ -64,13 +64,14 @@ int SerialPortBridgeGenericClass = core::RegisterObject("Send data (ex: force, d
 SerialPortBridgeGeneric::SerialPortBridgeGeneric()
     : d_port(initData(&d_port, "port", "Serial port name"))
     , d_baudRate(initData(&d_baudRate, "baudRate", "Transmission speed"))
-    , d_packetOut(initData(&d_packetOut, "sentData", "Data to send: the value 245 will be set at the beginning of the sent data as a header,\n"
+    , d_packetOut(initData(&d_packetOut, "sentData", "Data to send: the value 245 will be sent at the beginning of the sent data as a header,\n"
                            "enabling to implement a header research in the 'receiving' code, for synchronization purposes."))
-    , d_packetIn(initData(&d_packetIn, "receivedData", "Data received: vector of unsigned char, each entry should be an integer between 0 and 244."))
+    , d_packetIn(initData(&d_packetIn, "receivedData", "Data received: vector of unsigned char, each entry should be an integer between 0 and 255."))
```

Comment:
What happen if we send 245 as a value for command?  
final `sentData = [245,0,0,0, 245,10,10,10]`

Here's a typical Arduino code:
```
if (Serial.available())    
{    
    header = Serial.read();  
    while (header != 245)   
    {  
      while (!Serial.available());  
      header = Serial.read();  
    }  
    ...  
}
```
I see no reason why it couldn't start from the middle of the "sentData", right?

Reply:
You are right Eulalie, 

the wonderful world of serial communication protocols :)

Reply:
Wouldn't it be more convenient to say (and test) that the command should be contained between 0 and 244?  We will lose only 11 more possibilities... but guarantee less synchronization problem, isn't it?  Or maybe add an additional test in every arduino code is enough... I don't know.

Reply:
To me the data protocol in its current state is not failure proof so I support the idea of fixing it. 
The most important thing is to be sure that if an error occurs it has the only consequence on the current data frame and will not propagate (which can happen with the current one if one marker is skipped and one of the data contains 245). 

Doing what Eulalie suggest is fine and backward compatible. Using 255 instead of 245 only lose one value.

Reply:
I didn't want to change 245 by 255 because of the existing demos (specially the untracked ones), but I too think it's cleaner this way. If you agree Mario I can make the changes.

Reply:
Maybe you can add a protocol data fields to specify if you want the "old" 245 protocol or a new one.